library/windows/scripts/build_jsonlib.bat

```diff
@@ -0,0 +1,100 @@
+:: Copyright 2018 Google LLC
+::
+:: Licensed under the Apache License, Version 2.0 (the "License");
+:: you may not use this file except in compliance with the License.
+:: You may obtain a copy of the License at
+::
+::      http://www.apache.org/licenses/LICENSE-2.0
+::
+:: Unless required by applicable law or agreed to in writing, software
+:: distributed under the License is distributed on an "AS IS" BASIS,
+:: WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+:: See the License for the specific language governing permissions and
+:: limitations under the License.
+@echo off
+
+:: Find where VS lives and start a VC command prompt
+set pre=Microsoft.VisualStudio.Product.
+set ids=%pre%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+ 
+pushd "C:\Program Files (x86)\Microsoft Visual Studio\Installer\"
+for /f "usebackq tokens=1* delims=: " %%i in (`vswhere -latest -products *`) do (if /i "%%i"=="installationPath" set InstallDir=%%j)
+popd
+
+pushd %InstallDir%\VC\Auxiliary\Build
+call vcvarsall.bat x86_amd64
+popd
+
+set DEPENDDIREXISTS=true
+if not exist %~dp0..\dependencies\json\allocator.h set DEPENDDIREXISTS=false

```

Comment:
This is pretty fragile as a test. I just pulled this branch on Windows and tried building it, and didn't notice among other the failures I had (some just local configuration problems that aren't necessarily related to the PR), jsoncpp didn't actually build successfully.

Subsequent runs just print "jsoncpp found" and proceed, then the FDE build fails because the jsoncpp library isn't present, and I had to dig into the script to figure out that I had to nuke the source in order to get it to try building again.

(And looking more closely while debugging this, it's not clear to me why this variable is called DEPENDDIREXISTS when it's actually checking for the source having been downloaded, rather than the directory existing)